From 30995b35d1c36f12f9658883b0d5a168eb678139 Mon Sep 17 00:00:00 2001
From: Ken Zhu <quic_guigenz@quicinc.com>
Date: Fri, 29 Apr 2022 10:56:40 -0700
Subject: [PATCH] net: fix spin_lock issue

br_fdb_entry_refresh could be called under spin_lock protection
and spin_lock is not needed here.

Change-Id: I9333fb0ca1e1ab096239b06f72b5d2f722bce26e
Signed-off-by: Ken Zhu <quic_guigenz@quicinc.com>
---
 net/bridge/br_fdb.c | 2 --
 1 file changed, 2 deletions(-)

--- a/net/bridge/br_fdb.c
+++ b/net/bridge/br_fdb.c
@@ -720,12 +720,10 @@ void br_fdb_entry_refresh(struct net_dev
 		return;
 
 	rcu_read_lock();
-	spin_lock_bh(&p->br->hash_lock);
 	fdb = fdb_find_rcu(&p->br->fdb_hash_tbl, addr, vid);
 	if (likely(fdb)) {
 		fdb->updated = jiffies;
 	}
-	spin_unlock_bh(&p->br->hash_lock);
 	rcu_read_unlock();
 }
 EXPORT_SYMBOL_GPL(br_fdb_entry_refresh);
